File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix C/CS/DataServiceWebApplicationCS/DataServiceWebApplicationCS/DataBindingWebService.asmx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Services;
using System.Data;
using System.Data.SqlClient;

namespace DataServiceWebApplicationCS
{
    /// <summary>
    /// Summary description for DataBindingWebService
    /// </summary>
    [WebService(Namespace = "http://tempuri.org/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [System.ComponentModel.ToolboxItem(false)]
    // To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line. 
    // [System.Web.Script.Services.ScriptService]
    public class DataBindingWebService : System.Web.Services.WebService
    {

        [WebMethod]
        public DataSet GetData()
        {
            System.Data.SqlClient.SqlConnection sqlConnection1 = new System.Data.SqlClient.SqlConnection();
            sqlConnection1.ConnectionString = "Data Source=ANAMIKA-PC\\SQLEXPRESS;Initial Catalog=northwind;Integrated Security=True";
            string selectStr = "SELECT EmployeeID, FirstName, LastName,  City, Country FROM Employees";
            SqlDataAdapter da = new SqlDataAdapter(selectStr, sqlConnection1);
            DataSet ds = new DataSet();
            da.Fill(ds);
            return (ds);
        }
    }
}

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix C/VB/DataServiceWebApplicationVB/DataServiceWebApplicationVB/DataBindingWebService.asmx.vb
Imports System.Web.Services
Imports System.Web.Services.Protocols
Imports System.ComponentModel
Imports System.Data
Imports System.Data.SqlClient


' To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line.
' <System.Web.Script.Services.ScriptService()> _
<System.Web.Services.WebService(Namespace:="http://tempuri.org/")> _
<System.Web.Services.WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
<ToolboxItem(False)> _
Public Class DataBindingWebService
    Inherits System.Web.Services.WebService

    <WebMethod()> _
    Public Function GetData() As DataSet
        Dim sqlConnection1 As New System.Data.SqlClient.SqlConnection()
        sqlConnection1.ConnectionString = "Data Source=ANAMIKA-PC\SQLEXPRESS;Initial Catalog=northwind;Integrated Security=True"
        Dim selectStr As String = "SELECT EmployeeID, FirstName, LastName,  City, Country FROM Employees"
        Dim da As New SqlDataAdapter(selectStr, sqlConnection1)
        Dim ds As New DataSet()
        da.Fill(ds)
        Return (ds)
    End Function
End Class